### PR TITLE
Wayland & QoL improvements

### PR DESCRIPTION
**Pull request by r2binx**
**Changes in summary:**
- Allow user to disable Wayland support flags by environment variable
- Fix cursor
- Move command to separate script file in repo

### CODE REVIEW

**Comment by excieve** [comment]
This looks very useful, any chance for this to get merged? Slack runs much better in pure Wayland mode on nvidia than it does in XWayland. This would make enabling it more straightforward.

File: com.slack.Slack.json [hunk 1]
```diff
@@ -20,6 +21,7 @@
         "--filesystem=xdg-download",
         "--talk-name=org.freedesktop.Notifications",
         "--env=XDG_CURRENT_DESKTOP=Unity",
+        "--env=USE_WAYLAND=0",
```
Comment (joshua-stone) [inline com.slack.Slack.json:24]: This is needed to fix cursor themes when running on Wayland:

```
--env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
```

File: .github/workflows/flatpak-builder.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+              with:
+                  bundle: com.slack.Slack.flatpak
+                  manifest: com.slack.Slack.json
+                  cache-key: flatpak-builder-${{ github.sha }}
+                  arch: "x86_64"
+                  verbose: true
+            - name: Release
+              uses: softprops/action-gh-release@v1
+              with:
+                  files: com.slack.Slack.flatpak
```
Comment (bilelmoussaoui) [inline .github/workflows/flatpak-builder.yml:27]: What is the point of this when flathub has a buildbot already??

Comment (r2binx) [reply .github/workflows/flatpak-builder.yml]: Reverted. Sorry, I was testing some stuff and this wasn't meant to be here.

**Comment by goetzc** [comment]
Hi @r2binx. Why do you close this?

**Comment by r2binx** [comment]
> Hi @r2binx. Why do you close this?

For some reason it's not working for me anymore. Unfortunately, I don't have the time to debug it right now.
Also, removing the Wayland socket in e.g. Flatseal currently achieves the same thing.
As for the cursor size, I think it should be fixed in a separate PR.

**Pull request closed without merging**